dask/multiprocessing.py

```diff
@@ -4,9 +4,27 @@
 from .optimize import fuse, cull
 import multiprocessing
 import dill
+import cloudpickle
 import pickle
 from .async import get_async # TODO: get better get
 from .context import _globals
+from sys import getrecursionlimit, setrecursionlimit
+
+def _dumps(x):
+    try:
+        return cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+    except:
+        try:
+            recursionlimit = getrecursionlimit()
+            setrecursionlimit(2*recursionlimit)
+            dumped = cloudpickle.dumps(x, protocol=pickle.HIGHEST_PROTOCOL)
+            setrecursionlimit(recursionlimit)
+            return dumped
+        except:
+            raise
```

Comment:
It seems like the probem occurs for method descriptor objects:

    Python 2.7.10 |Continuum Analytics, Inc.| (default, Oct 19 2015, 18:04:42) 
    [GCC 4.4.7 20120313 (Red Hat 4.4.7-1)] on linux2
    Type "help", "copyright", "credits" or "license" for more information.
    Anaconda is brought to you by Continuum Analytics.
    Please check out: http://continuum.io/thanks and https://anaconda.org
    >>> import cloudpickle, pickle
    
    >>> cloudpickle.dumps(set.union)
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/site-packages/cloudpickle/cloudpickle.py", line 602, in dumps
        cp.dump(obj)
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/site-packages/cloudpickle/cloudpickle.py", line 111, in dump
        raise pickle.PicklingError(msg)
    pickle.PicklingError: Could not pickle object as excessively deep recursion required.
    
    >>> cloudpickle.dumps(str.decode)
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/site-packages/cloudpickle/cloudpickle.py", line 602, in dumps
        cp.dump(obj)
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/site-packages/cloudpickle/cloudpickle.py", line 111, in dump
        raise pickle.PicklingError(msg)
    pickle.PicklingError: Could not pickle object as excessively deep recursion required.
    
    >>> pickle.dumps(set.union)
    Traceback (most recent call last):
      File "<stdin>", line 1, in <module>
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/pickle.py", line 1374, in dumps
        Pickler(file, protocol).dump(obj)
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/pickle.py", line 224, in dump
        self.save(obj)
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/pickle.py", line 306, in save
        rv = reduce(self.proto)
      File "/home/pmd/anaconda3/envs/python2/lib/python2.7/copy_reg.py", line 70, in _reduce_ex
        raise TypeError, "can't pickle %s objects" % base.__name__
    TypeError: can't pickle method_descriptor objects
    
    >>> def f(x):
    ...     return set.union(x)
    ... 
    >>> cloudpickle.dumps(f)
    '\x80\x02ccloudpickle.cloudpickle\n_fill_function\nq\x00(ccloudpickle.cloudpickle\n_make_skel_func\nq\x01ccloudpickle.cloudpickle\n_builtin_type\nq\x02U\x08CodeTypeq\x03\x85q\x04Rq\x05(K\x01K\x01K\x02KCU\rt\x00\x00j\x01\x00|\x00\x00\x83\x01\x00Sq\x06N\x85q\x07U\x03setq\x08U\x05unionq\t\x86q\nU\x01xq\x0b\x85q\x0cU\x07<stdin>q\rU\x01fq\x0eK\x01U\x02\x00\x01q\x0f))tq\x10Rq\x11]q\x12}q\x13\x87q\x14Rq\x15}q\x16N}q\x17tR.'
    
    >>> def g(x):
    ...     return str.decode(x)
    ... 
    >>> cloudpickle.dumps(g)
    '\x80\x02ccloudpickle.cloudpickle\n_fill_function\nq\x00(ccloudpickle.cloudpickle\n_make_skel_func\nq\x01ccloudpickle.cloudpickle\n_builtin_type\nq\x02U\x08CodeTypeq\x03\x85q\x04Rq\x05(K\x01K\x01K\x02KCU\rt\x00\x00j\x01\x00|\x00\x00\x83\x01\x00Sq\x06N\x85q\x07U\x03strq\x08U\x06decodeq\t\x86q\nU\x01xq\x0b\x85q\x0cU\x07<stdin>q\rU\x01gq\x0eK\x01U\x02\x00\x01q\x0f))tq\x10Rq\x11]q\x12}q\x13\x87q\x14Rq\x15}q\x16N}q\x17tR.'
    >>>